modules/material/material.service.js

```diff
@@ -32,11 +36,46 @@ class MaterialsService {
     return await Material.findByIdAndUpdate(id, material, { new: true });
   }
 
-  async addMaterial(data) {
-    if (await this.checkMaterialExist(data)) {
+  async addMaterial({ material, images }) {
+    const { additionalPrice, ...rest } = material;
+
+    const currency = await Currency.findOne();
```

Comment:
Place, move currency declaration after validation checks.
There is no need in 'currency' const if material exist or no images.

Reply:
fixed